src/test/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollectorTest.java

```diff
@@ -503,6 +503,25 @@ void testGuestCountIsTrackedOnHost() {
         assertEquals(2, savedHypervisor.getNumOfGuests().intValue());
     }
 
+    @Test
+    public void testTotalHosts() {
+        List<String> targetAccounts = Arrays.asList("A1");
+
+        InventoryHostFacts hypervisor = createHypervisor("A1", "O1", NON_RHEL_PRODUCT_ID, 12, 3);
+
+        Map<String, String> expectedHypervisorMap = new HashMap<>();
+        expectedHypervisorMap.put(hypervisor.getSubscriptionManagerId(),
+            hypervisor.getSubscriptionManagerId());
+        mockReportedHypervisors(targetAccounts, expectedHypervisorMap);
+
+        when(inventoryRepo.getFacts(eq(targetAccounts), anyInt()))
+                .thenReturn(Arrays.asList(hypervisor).stream());
+
+        Map<String, AccountUsageCalculation> calcs = collector.collect(NON_RHEL_PRODUCTS, targetAccounts);
+        assertEquals(1, calcs.size());
+        assertEquals(1, collector.getTotalHosts().count());
```

Comment:
Rather than assert the count as `1`, it looks like you'll need to capture the counter value before `collect`, and then assert that the diff is `1`. Otherwise, other tests that run before this one invalidate the test. :-)

Reply:
Thanks, that solves a problem I was having with this test.